server/src/controllers/order.controller.ts

```diff
@@ -54,39 +57,53 @@ const OrderController = {
     } catch (error) {
       return res.status(500).json({ message: error.message });
     }
+  },
+
+  async payAmount(req: Request, res: Response): Promise<Response> {
+    let isPaymentSucess = "1";
+    try {
+      request(
+        {
+          method: "POST",
+          url: `https://${process.env.key_id}:${process.env.key_secret}@api.razorpay.com/v1/payments/${req.params.paymentId}/capture`,
+          form: {
+            amount: req.body.total,
+            currency: "INR"
+          }
+        },
+        async (error: any, response: request.Response) => {
+          if (error) isPaymentSucess = "0";
+          else {
+            const userId = req.body.user._id;
+            const user = await User.findById(userId);
```

Comment:
Since we are updating user, this should be `let user = ...`

Reply:
Thanks missed on this crucial point. However my user did get updated while testing.